src/net/wooga/jenkins/pipeline/BuildUtils.groovy

```diff
@@ -3,19 +3,47 @@ package net.wooga.jenkins.pipeline
 import org.jenkinsci.plugins.pipeline.modeldefinition.Utils
 
 class BuildVersion {
+
     final String version
     final Boolean optional
+    // NET_4_6, NET_Standard_2_0 (DEFAULT)
+    final String apiCompatibilityLevel
 
-    BuildVersion(version, optional) {
+    BuildVersion(version, optional, apiCompatibilityLevel) {
         this.version = version
         this.optional = optional
+        this.apiCompatibilityLevel = apiCompatibilityLevel
     }
 
     @Override
     @NonCPS
     String toString() {
         return version
     }
+
+    @NonCPS
+    String toLabel(){
+        def result = version
+        if (optional){
+            result += " (optional)"
+        }
+        if (apiCompatibilityLevel != null) {
+            result += " (${apiCompatibilityLevel})"
+        }
+        return result
+    }
+
+    @NonCPS
+    String toDirectoryName(){
```

Comment:
That is a very specific name for this method which only formats the `BuildVersion` to a specific String. Not that this is not a good thing but I would suggest to move these kinds of methods to another object which actually needs them. Here the pipeline itself. It should not be part of the helper object or model in this case to provide format methods for all kinds of usecases.

Reply:
I see your point; in this case I thought having the method with the object would be useful in case we wanted to use this class in other places (as previously discussed), since using the build version to name the directory is a common use case.

Reply:
Yes I understand. It's mainly that formating strings is very opinionated based on the implementation details. We can leave it here. But any code that is added to objects bring more potential public APIs. If the formating would happend internally then we can change it without any issues.

Reply:
Agreed.